src/lib/features/download-episode/lib/specs/download.spec.ts

```diff
@@ -1,28 +1,35 @@
+import fetch from 'jest-mock-fetch';
 import { get } from 'svelte/store';
+import { waitFor } from '@testing-library/svelte';
+import { ReadableStream } from 'web-streams-polyfill/ponyfill';
 import { startEpisodeDownload } from '../download';
 
-jest.useFakeTimers();
-jest.spyOn(global, 'setTimeout');
+afterEach(fetch.reset);
+const exampleUrl = 'https://example.com/';
 
-const mockFetchAndWatchProgress = jest
-  .fn()
-  .mockName('fetchAndWatchProgress() from $lib/features/download-episode/lib')
-  .mockImplementation(async (url, store) => {
-    store.set(0);
-    jest.advanceTimersByTime(1000);
-    store.set(100);
-  });
-
-jest.mock('../download', () => ({
-  ...jest.requireActual('../download'),
-  fetchAndWatchProgress: () => mockFetchAndWatchProgress,
-}));
+it('constructs the correct URL (appends ?download=true)', () => {
+  startEpisodeDownload(exampleUrl);
+  expect(fetch).toHaveBeenCalledWith(`${exampleUrl}?download=true`);
+});
 
-it.skip('starts downloading episodes and watches progress', async () => {
-  const exampleUrl = 'https://example.com';
+it('watches the download progress', async () => {
+  const sampleResponseBytes = [1, 2, 3];
   const store = startEpisodeDownload(exampleUrl);
-  expect(mockFetchAndWatchProgress).toHaveBeenCalledWith(`${exampleUrl}?download=true`, store);
+  let streamController!: ReadableStreamDefaultController<Uint8Array>;
```

Comment:
If this stream controller is being used only inside the `ReadableStream` constructor and not for, say, `expect`ing anything on it, why does it need to be stored outside?

I tried removing this variable and changing the constructor to use
```ts
start(controller) {
  controller.enqueue(new Uint8Array(sampleResponseBytes));
  controller.close();
},
```
and it still worked fine

Reply:
Oh wait, I think I got it. You wanted to make sure the `enqueue`ing doesn't happen before you check for the store containing `0` to make sure no race conditions occur. Is that right?

Reply:
Yes